docs/model-dev-guide/dtrain/dtrain-implement.rst

```diff
@@ -24,110 +25,117 @@ training, as described in :ref:`cluster-configuration`.
-``global_batch_size // slots_per_trial``. The per-slot and global batch sizes should be accessed via
-the context using :func:`context.get_per_slot_batch_size()
+``global_batch_size // slots_per_trial``. The per-slot and global batch sizes can be accessed
+through the context using :func:`context.get_per_slot_batch_size()
 <determined.TrialContext.get_per_slot_batch_size>` and :func:`context.get_global_batch_size()
 <determined.TrialContext.get_global_batch_size>`, respectively. If ``global_batch_size`` is not
 evenly divisible by ``slots_per_trial``, the remainder is dropped.
 
-If :ref:`slots_per_trial <exp-config-resources-slots-per-trial>` is greater than the number of slots
-on a single agent, Determined schedules it over multiple machines. When scheduling a multi-machine
-distributed training job, Determined requires that the job uses all of the slots (GPUs) on an agent.
-For example, in a cluster that consists of 8-GPU agents, an experiment with :ref:`slots_per_trial
-<exp-config-resources-slots-per-trial>` set to ``12`` is never scheduled and will wait indefinitely.
-The section on :ref:`Scheduling Behavior <dtrain-scheduling>` describes this in more detail.
+If the value of :ref:`slots_per_trial <exp-config-resources-slots-per-trial>` is greater than the
+number of slots available on a single agent, Determined schedules it over multiple machines. When
+scheduling a multi-machine distributed training job, Determined requires that all slots (GPUs) on an
+agent are used by the job. For example, in a cluster composed of 8-GPU agents, an experiment that
+has :ref:`slots_per_trial <exp-config-resources-slots-per-trial>` set to ``12`` will not be
+scheduled and will wait indefinitely. For more details, you can visit :ref:`Scheduling Behavior
+<dtrain-scheduling>`.
 
-There might also be running tasks preventing your multi-GPU trials from acquiring enough GPUs on a
-single machine. Consider adjusting ``slots_per_trial`` or terminating existing tasks to free slots
-in your cluster.
+You might have existing tasks that are running on a single machine that are preventing your
+multi-GPU trials from acquiring sufficient GPUs. To alleviate this, you may want to consider
+adjusting ``slots_per_trial`` or terminating existing tasks to free up slots in your cluster.
 
 Global Batch Size
 =================
 
-When doing distributed training, the ``global_batch_size`` specified in the
+During distributed training, the ``global_batch_size`` specified in the
 :ref:`experiment-config-reference` is partitioned across ``slots_per_trial`` GPUs. The per-GPU batch
 size is set to: ``global_batch_size // slots_per_trial``. If ``slots_per_trial`` does not divide
 ``global_batch_size`` evenly, the remainder is dropped. For convenience, the per-GPU batch size can
 be accessed via the Trial API, using :func:`context.get_per_slot_batch_size
 <determined.TrialContext.get_per_slot_batch_size>`.
```

Comment:
We probably should insert an explicit caveat somewhere that this is a Trial API-only feature and that with Core API the user needs to handle batch size configuration manually.

Reply:
@garrett361 there is now a note added to the Advanced Optimizations section, please advise if this note should be in other sections as well.

Note

These optimizations only apply to [High-Level APIs](http://localhost:1234/model-dev-guide/apis-howto/overview.html#high-level-apis) (Trial APIs) and do not apply to the Core API.

Reply:
i had the note in the wrong section. moved it under Global Batch Size.